src/results.cpp

```diff
@@ -726,6 +731,8 @@ void Results::prepare_async(ForCallback force)
 
 NotificationToken Results::add_notification_callback(CollectionChangeCallback cb) &
 {
+    if (m_descriptor_ordering.will_apply_limit())
+        throw UnimplementedOperationException("Change notifications for Results with a limit are not yet implemented");
```

Comment:
I don't know the notification code at all, but I'm a bit surprised to see this will need special handling.